pkg/kv/kvserver/store_snapshot.go

```diff
@@ -1189,29 +1177,6 @@ var rebalanceSnapshotRate = settings.RegisterByteSizeSetting(
 	},
 ).WithPublic()
 
-// recoverySnapshotRate is the rate at which Raft-initiated snapshot can be
-// sent. Ideally, one would never see a Raft-initiated snapshot; we'd like all
-// replicas to start out as learners or via splits, and to never be cut off from
-// the log. However, it has proved unfeasible to completely get rid of them.
-//
-// TODO(tbg): The existence of this rate, separate from rebalanceSnapshotRate,
-// does not make a whole lot of sense. Both sources of snapshots compete thanks
-// to a semaphore at the receiver, and so the slower one ultimately determines
-// the pace at which things can move along.
-var recoverySnapshotRate = settings.RegisterByteSizeSetting(
-	settings.SystemOnly,
-	"kv.snapshot_recovery.max_rate",
-	"the rate limit (bytes/sec) to use for recovery snapshots",
-	32<<20, // 32mb/s
-	func(v int64) error {
-		if v < minSnapshotRate {
-			return errors.Errorf("snapshot rate cannot be set to a value below %s: %s",
-				humanizeutil.IBytes(minSnapshotRate), humanizeutil.IBytes(v))
-		}
-		return nil
-	},
-).WithPublic()
```

Comment:
Shouldn't we "Retire" that metric before just ripping it out?

https://github.com/cockroachdb/cockroach/blob/b39f02aaf2a0ea4fe93a26a22d5c4d82943a8340/pkg/settings/common.go#L96-L101

Reply:
Here's an end to end example https://github.com/cockroachdb/cockroach/blob/fb091771a13c7104678e76942ac055da8fd7ecaf/pkg/ts/db.go#L63-L72

Reply:
Thanks! I changed to retired. and cleaned it up. It ended up being much better to use in the init function as it doesn't assign it a variable name (it is provably retired, not just a naming convention).